src/ActionModal/utils/ActionManager.js

```diff
@@ -149,20 +149,24 @@ function convertCurrencyData(amounts, network) {
 
 // limitation of the Ledger Nano S, so we pick the top 5 rewards and inform the user.
 export function getTop5RewardsValidators(rewards) {
-  const mostClaimableToken = rewards.reduce((topRewardToken, reward) => {
-    if (Number(reward.amount) > Number(topRewardToken.amount)) {
-      return reward
+  const rewardsPerValidatorObject = rewards.reduce((all, reward) => {
+    return {
+      ...all,
+      [reward.validator.operatorAddress]:
+        Number(reward.amount) +
+        (Number(all[reward.validator.operatorAddress]) || 0)
     }
-    return topRewardToken
+  }, {})
+  const rewardsPerValidatorAddresses = Object.keys(rewardsPerValidatorObject)
+  let rewardsPerValidatorArray = []
+  rewardsPerValidatorAddresses.forEach((validatorAddress, index) => {
+    rewardsPerValidatorArray.push({
+      validator: validatorAddress,
+      totalRewardAmount: Object.values(rewardsPerValidatorObject)[index]
+    })
```

Comment:
Do you know about `Object.entries`?

Reply:
I should since I copied it from you in my PR 🤣 

But yeah, here it doesn't work since they don't have "keys" (so I cannot do:
```
const rewardsPerValidatorArray = Object.entries(rewardsPerValidatorObject)
rewardsPerValidatorArray.map(([a, b]) => {a, b})
```
, since rewardsPerValidatorObject just has <validatorAddress>:<totalRewardsAmoun>, and no "a" nor "b" key or none at all for that matter)

Reply:
```
const rewardsPerValidatorArray = Object.entries(rewardsPerValidatorObject)
rewardsPerValidatorArray.map(([validator, totalRewardAmount]) => ({validator, totalRewardAmount}))
```

Reply:
Yes, but for that I needed this forEach

Reply:
where do you see a forEach in my code? ^^

Reply:
I mean in my code xD

this one:
```
  rewardsPerValidatorAddresses.forEach((validatorAddress, index) => {
    rewardsPerValidatorArray.push({
      validator: validatorAddress,
      totalRewardAmount: Object.values(rewardsPerValidatorObject)[index]
    })
```
This is where I introduce those keys. They weren't there before

Reply:
noooo
check my code...
the "keys" are just names you give to the "first" and "second" item of the array which is the first argument of the funtion. It's called array destructuring: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

Reply:
![image](https://user-images.githubusercontent.com/40721795/78191370-22fa1880-7476-11ea-9780-7e5763fed050.png)

It doesn't work 🤷‍♀ 

Maybe we can go through this in the next 1:1?

Reply:
you forgot the `()`. when you return an object from an arrow function you need to do `() => ({})`

Reply:
🤦‍♀ something so simple!

Wow, thank you so much for going through this with me Fabo! 😸  It was sure an important lesson